Core/StorageCache.swift

```diff
@@ -91,29 +57,22 @@ public class StorageCache: StorageCacheUpdating {
             HTTPSUpgrade.shared.loadData()
             return result
             
-        case .surrogates:
-            guard let data = data as? Data else { return false }
-            return surrogateStore.parseAndPersist(data: data)
+        case .httpsBloomFilterSpec:
+            return false
             
-        case .entitylist:
-            guard let data = data as? Data else { return false }
-            let result = entityMappingStore.persist(data: data)
-            entityMapping = EntityMapping(store: entityMappingStore)
-            return result
+        case .surrogates:
+            return fileStore.persist(data as? Data, forConfiguration: configuration)
             
-        default:
+        case .trackerDataSet:
+            if fileStore.persist(data as? Data, forConfiguration: configuration) {
+                TrackerDataManager.shared.reload()
```

Comment:
Basically if we are reloading here, then we should not be left with embedded data set - I'd make it so `reload()` method returns what kind of data it uses (fetched/embedded) and fire a pixel if we do not manage to load what we have just persisted.